shared/teams/team/settings/retention/index.js

```diff
@@ -62,21 +74,51 @@ class RetentionPicker extends React.Component<Props, State> {
       // noop
       return
     }
-    const onConfirm = () => this.props.setRetentionPolicy(selected)
+    const onConfirm = () => {
+      this.props.setRetentionPolicy(selected)
+    }
     const onCancel = this._init
     if (decreased) {
       // show warning
+      this._showSaved = false
       this.props.onShowWarning(policyToDays(selected, this.props.teamPolicy), onConfirm, onCancel)
       return
     }
     // set immediately
     onConfirm()
+    this._showSaved = true
+    this._setSaving()
   }
 
   _onSelect = (selected: RetentionPolicy) => {
+    // set saveState to 'same' in case we're in the middle of an animation
+    this._setSame()
     this.setState({selected}, this._handleSelection)
   }
 
+  _setSaving = () => {
+    this.setState({saveState: 'saving'})
+  }
+
+  _setJustSaved = () => {
+    if (!this._showSaved) {
+      // Don't show if we went through the confirmation dialog flow
+      this._setSame()
+      return
+    }
+    this.setState({saveState: 'justSaved'})
+    this.props.clearTimeout(this._timeoutID)
```

Comment:
I think we can factor out the timeout logic from the notifications component pretty cleanly, just so we won't have to keep reimplementing timeout logic and we can perhaps reuse this for other screens that need a similar save state component.

I'm envisioning it being used something like

```
render() {
...
  <SaveIndicator state={this.state._saveState} minSavingTimeMs={300} savedTimeoutMs={2500} />
...
}
```
where the outer component will pass in the two time interval props, and `_saveState` (of type `SaveStateType`), and the component will detect when `_saveState` changes and handle all the timing logic.

What do you think?

Reply:
That'd be cool! I don't think I want to include it in this PR since it seems nontrivial to take out of `Notifications`, and I don't want this to get much bigger. Do you want to ticket it?

Reply:
Okay, I'll ticket it!